File: cpp/1258.cpp
#include <iostream>
#include <string>
#include <map>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int n, i;
    string name, uniform;
    cin >> n;
    getchar();

    while (n != 0) {
        map<int, vector<string>> uniforms;
        string size_color[6] = {"branco P", "branco M", "branco G", "vermelho P", "vermelho M", "vermelho G"};

        for (i = 0; i < n; i++) {
            getline(cin, name);
            getline(cin, uniform);
            auto itr = find(size_color, size_color + 6, uniform);
            uniforms[distance(size_color, itr)].push_back(name);
        }

        for (auto key: uniforms) {
            sort(key.second.begin(), key.second.end());

            for (auto element: key.second) {
                cout << size_color[key.first] << ' ' << element << '\n';
            }
        }
        
        uniforms.clear();
        cin >> n;
        getchar();

        if (n != 0) {
            cout << '\n';
        }
    }

    return 0;
}

File: py/1258.py
def tshirts(n):
    for key in n:
        n[key].sort()
        
        for element in n[key]:
            print(key, element)

n = int(input())
while True:
    uniforms = {
        'branco P': [],
        'branco M': [],
        'branco G': [],
        
        'vermelho P': [],
        'vermelho M': [],
        'vermelho G': [],
    }
    
    while n > 0:
        ps = (input(), input())
        uniforms[ps[1]].append(ps[0])
        
        n -= 1
        
    tshirts(uniforms)

    n = int(input())
    
    if n != 0:
        print('')
    else:
        break